addons/mrp/models/mrp_workorder.py

```diff
@@ -782,28 +782,28 @@ def _get_duration_expected(self, alternative_workcenter=False, ratio=1):
         self.ensure_one()
         if not self.workcenter_id:
             return self.duration_expected
+        capacity, setup, cleanup = self.workcenter_id._get_capacity(self.product_id, self.product_uom_id, self.production_bom_id.product_qty or 1)
         if not self.operation_id:
-            duration_expected_working = (self.duration_expected - self.workcenter_id.time_start - self.workcenter_id.time_stop) * self.workcenter_id.time_efficiency / 100.0
+            duration_expected_working = (self.duration_expected - setup - cleanup) * self.workcenter_id.time_efficiency / 100.0
             if duration_expected_working < 0:
                 duration_expected_working = 0
             if self.qty_producing not in (0, self.qty_production, self._origin.qty_producing):
                 qty_ratio = self.qty_producing / (self._origin.qty_producing or self.qty_production)
             else:
                 qty_ratio = 1
-            return self.workcenter_id._get_expected_duration(self.product_id) + duration_expected_working * qty_ratio * ratio * 100.0 / self.workcenter_id.time_efficiency
-        qty_production = self.production_id.product_uom_id._compute_quantity(self.qty_producing or self.qty_production, self.production_id.product_id.uom_id)
-        capacity = self.workcenter_id._get_capacity(self.product_id)
+            return setup + cleanup + duration_expected_working * qty_ratio * ratio * 100.0 / self.workcenter_id.time_efficiency
+        qty_production = self.qty_producing or self.qty_production
         cycle_number = float_round(qty_production / capacity, precision_digits=0, rounding_method='UP')
         if alternative_workcenter:
             # TODO : find a better alternative : the settings of workcenter can change
-            duration_expected_working = (self.duration_expected - self.workcenter_id._get_expected_duration(self.product_id)) * self.workcenter_id.time_efficiency / (100.0 * cycle_number)
+            duration_expected_working = (self.duration_expected - setup - cleanup) * self.workcenter_id.time_efficiency / (100.0 * cycle_number)
```

Comment:
self.duration_expected is losing the alternative context no?

Reply:
no, as we are retrieving capacity, setup & cleanup on alternative_workcenter
a few lines below to calculate cycle_number and correct duration
this is to get the normal operation duration without any workcenter context